### DISCUSSION TITLE
How to use my custom domain to access my organization like - [domain].github.com?

### DISCUSSION THREAD

**Discussion by nikhil-veditr** [community/community#49329] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have searched everywhere but couldn't find anything regarding it. So, basically I want to use my domain to use my organization like some organization does. If anybody has something which can help me, please let me know.

**Comment by rauzerfabiano** [comment]
To use your custom domain to access your organization on GitHub, you will need to set up a CNAME record in your domain's DNS settings. Here are the steps to follow:

    Go to your domain registrar's website and log in to your account.
    Navigate to your domain's DNS settings or Zone Editor.
    Add a new CNAME record with the following settings:

    Hostname or Alias: [domain].github.io (replace [domain] with your actual domain name)
    Points to or Value: [your organization name].github.io (replace [your organization name] with your actual organization name)

    Save the changes to your DNS settings.

After you have set up the CNAME record, it may take some time for the changes to propagate and for your custom domain to start working. Once the changes have propagated, you should be able to access your organization on GitHub using your custom domain like this: [domain].github.com.

**Comment by nikhil-veditr** [comment]
Thank you very much 🥰. Let me try once.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by nikhil-veditr** [comment]
Not working. After adding CNAME record to the DNS record I am not able to access [domain].github.com or [domain].github.io. Can you please give me one example with an actual domain like google.com how to add the record if I am adding it wrong?
________________________________

**Comment by vitoriazoche** [comment]
Have you tried this https://stackoverflow.com/questions/66001698/how-to-make-a-github-pages-for-an-organization ?

**Comment by zenjichen** [comment]
You want [yourdomain].github.com but with your own domain. As in:
code.yourcoolstartup.com magically behaving like yourorg.github.com.

Short version: you can’t.

Longer, slightly more dignified version:

[org].github.com is controlled entirely by GitHub. That subdomain space belongs to them. You don’t get to plug your DNS into their naming system like it’s a USB port.

Now, what you can do depends on what you actually mean. And humans are notoriously vague about this stuff.

If You Mean: “I want a custom domain for GitHub Pages”

That is completely supported.

If your org has a repo like:

yourorg.github.io

You can:

Go to Repo → Settings → Pages

Add your custom domain (e.g., www.yourdomain.com)

Add a DNS record:

CNAME → yourorg.github.io

That gives you:

www.yourdomain.com

serving your GitHub Pages site.

This works beautifully. Very normal. Very civilized.

If You Mean: “I want my GitHub organization profile at my custom domain”

You cannot make:

yourdomain.com → github.com/yourorg

inside GitHub.

But you can redirect.

Just point your domain to:

https://github.com/yourorg

Using:

A DNS provider with URL forwarding

Cloudflare redirect rule

A tiny static page that redirects

It won’t change GitHub’s URL. It just forwards visitors.

If You Mean: “I want subdomains like repo.yourdomain.com mapping to GitHub repos”

That’s not a built-in GitHub feature.

You would need:

A reverse proxy

Or a custom frontend

Or GitHub Enterprise Server with custom domain configuration

On github.com (the public SaaS), you don’t get custom subdomain routing to orgs or repos.

Why This Exists

GitHub owns:

*.github.com

You own:

*.yourdomain.com

DNS does not let you stitch those namespaces together like Frankenstein’s branding experiment.

The Clean, Realistic Setup Most Teams Use

yourdomain.com → marketing site

code.yourdomain.com → redirect to GitHub org

docs.yourdomain.com → GitHub Pages

Repo access stays on github.com/yourorg

No one serious tries to re-skin GitHub’s core URLs. It’s unnecessary and brittle.

If you tell me what your actual goal is, we can design something that doesn’t fight how the internet works. Right now it sounds like you’re trying to repaint a rented apartment’s exterior. DNS does not negotiate with aesthetics.

**Comment by nikhil-veditr** [comment]
Ok got it, that’s fine, i wanted to know just what is possible with my requirement.


nikhil kamboj







---- On Mon, 02 Mar 2026 10:09:51 +0530 Hà Dopaaa ***@***.***> wrote ----





You want [yourdomain].github.com but with your own domain. As in:
 code.yourcoolstartup.com magically behaving like yourorg.github.com.
Short version: you can’t.

Longer, slightly more dignified version:

[org].github.com is controlled entirely by GitHub. That subdomain space belongs to them. You don’t get to plug your DNS into their naming system like it’s a USB port.

Now, what you can do depends on what you actually mean. And humans are notoriously vague about this stuff.

If You Mean: “I want a custom domain for GitHub Pages”

That is completely supported.

If your org has a repo like:

yourorg.github.io

You can:

Go to Repo → Settings → Pages

Add your custom domain (e.g., http://www.yourdomain.com )

Add a DNS record:

CNAME → yourorg.github.io

That gives you:

http://www.yourdomain.com 

serving your GitHub Pages site.

This works beautifully. Very normal. Very civilized.

If You Mean: “I want my GitHub organization profile at my custom domain”

You cannot make:

yourdomain.com → github.com/yourorg

inside GitHub.

But you can redirect.

Just point your domain to:

https://github.com/yourorg 

Using:

A DNS provider with URL forwarding

Cloudflare redirect rule

A tiny static page that redirects

It won’t change GitHub’s URL. It just forwards visitors.

If You Mean: “I want subdomains like repo.yourdomain.com mapping to GitHub repos”

That’s not a built-in GitHub feature.

You would need:

A reverse proxy

Or a custom frontend

Or GitHub Enterprise Server with custom domain configuration

On github.com (the public SaaS), you don’t get custom subdomain routing to orgs or repos.

Why This Exists

GitHub owns:

*.github.com

You own:

*.yourdomain.com

DNS does not let you stitch those namespaces together like Frankenstein’s branding experiment.

The Clean, Realistic Setup Most Teams Use

yourdomain.com → marketing site

code.yourdomain.com → redirect to GitHub org

docs.yourdomain.com → GitHub Pages

Repo access stays on github.com/yourorg

No one serious tries to re-skin GitHub’s core URLs. It’s unnecessary and brittle.

If you tell me what your actual goal is, we can design something that doesn’t fight how the internet works. Right now it sounds like you’re trying to repaint a rented apartment’s exterior. DNS does not negotiate with aesthetics.
